### PR TITLE
Project setup and build

### PR DESCRIPTION
**Pull request by beczkowb**

### CODE REVIEW

File: biggerquery/build.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+def build_dags(
+        root_package,
+        project_dir,
+        docker_repository,
+        start_time,
+        version,
+        specific_workflow=None):
+    for workflow in walk_workflows(root_package):
+        print(f'Generating DAG for {workflow.workflow_id}')
+        if specific_workflow is not None and specific_workflow != workflow.workflow_id:
```
Comment (iwko) [inline biggerquery/build.py:84]: if specific_workflow is None and specific_workflow == workflow.workflow_id:
          generate_dag_file(
            resolve(project_dir),
            docker_repository,
            workflow,
            start_time,
            version,
            resolve(root_package).split(os.sep)[-1])

?

File: biggerquery/build.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    Example:
+    from setuptools import setup
+    from biggerquery.build import project_setup, auto_configuration
+
+    setup(project_setup(**auto_configuration('my_super_project')))
+    '''
+    if project_name is None or docker_repository is None or root_package is None or project_dir is None or \
+            build_dir is None or test_package is None or dags_dir is None or dist_dir is None or image_dir is None or \
+            eggs_dir is None or deployment_config_file is None or version is None or resources_dir is None or project_requirements_file is None:
+        raise ValueError('You need to provide all parameters for the project_setup function.')
```
Comment (iwko) [inline biggerquery/build.py:268]: Could we provide explictly info about missing parameters?

File: biggerquery/version.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    latest_tag = get_tag()
+    if latest_tag:
+        tag = bump_minor(latest_tag)
+    else:
+        tag = '0.1.0'
+    git_tag_command(tag)
+
+
+def get_version():
+    # TODO dirty master protection
```
Comment (iwko) [inline biggerquery/version.py:31]: ?

File: test/example_project/deployment_config.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import os
+import sys
+from pathlib import Path
+
+
+def add_bigflow_to_path():
```
Comment (iwko) [inline test/example_project/deployment_config.py:6]: Is this used?

File: test/test_build.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+def create_image_leftovers(test_project_dir_path: Path = TEST_PROJECT_PATH):
+    mkdir(test_project_dir_path / 'image')
+    (test_project_dir_path / 'image' / 'leftover').touch()
+
+
+def create_package_leftovers(
+        test_project_dir_path: Path = TEST_PROJECT_PATH,
+        project_name: str = PROJECT_NAME):
+    mkdir(test_project_dir_path / 'build')
+    (test_project_dir_path / 'build' / 'leftover').touch()
```
Comment (iwko) [inline test/test_build.py:46]: could we move lines like this to mkdir method?

File: biggerquery/build.py (hunk 1 above)
Comment (beczkowb) [reply biggerquery/build.py:84]: When you specify a single workflow to generate the DAG, this if statement takes care of skipping the DAGs you didn't choose.

File: test/test_build.py (hunk 5 above)
Comment (beczkowb) [reply test/test_build.py:46]: The `touch` method creates a file, so we can't

File: test/example_project/deployment_config.py (hunk 4 above)
Comment (beczkowb) [reply test/example_project/deployment_config.py:6]: No

Comment (beczkowb) [reply test/example_project/deployment_config.py]: done

File: biggerquery/build.py (hunk 2 above)
Comment (beczkowb) [reply biggerquery/build.py]: done

File: biggerquery/version.py (hunk 3 above)
Comment (beczkowb) [reply biggerquery/version.py]: done

**Pull request merged**